src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3353,9 +3354,18 @@ node.Type switch
         /// <param name="depth">The depth of the current node.</param>
         private static void DescribeExpression(TextWriter writer, RegexNode node, string prefix, int depth = 0)
         {
-            bool skip =
-                node.Type is RegexNode.Concatenate ||
-                (node.Type == RegexNode.Atomic && node.Child(0).Type is RegexNode.Loop or RegexNode.Lazyloop or RegexNode.Alternate);
+            bool skip = node.Type switch
+            {
+                // For concatenations, flatten the contents into the parent, but only if the parent isn't a form of alternation,
+                // where each branch is considered to be independent rather than a concatenation.
+                RegexNode.Concatenate when node.Next is not { Type: RegexNode.Alternate or RegexNode.Testref or RegexNode.Testgroup } => true,
```

Comment:
NIT: is there a reason why here we use `node.Next` and bellow we use `node.Child(0)`? Or am I incorrectly assuming they are the same?

Reply:
Ironically, I just opened https://github.com/dotnet/runtime/issues/62651. :wink: `Next` should really be called `Parent`.